src/lbaf/Utils/lbsJSONTaskLister.py

```diff
@@ -0,0 +1,126 @@
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+# Questions? Contact darma@sandia.gov
+#
+###############################################################################
+#@HEADER
+#
+"""
+Utility to generate a yaml containing lists of tasks associated to their respective ranks,
+from the last phase and last sub-iteration of input JSON files.
+
+"""
+
+import os
+import json
+import yaml
+import argparse
+
+from lbaf.IO.lbsVTDataReader import LoadReader
+from lbaf.Utils.lbsLogging import get_logger, Logger
+
+class JSONTaskLister:
+    def __process_files(self):
+        reader = LoadReader(
+            file_prefix = self.__directory + self.__file_stem,
+            logger = self.__logger,
+            file_suffix = self.__file_suffix
+        )
+
+        tasks = {}
+        n_ranks = reader.n_ranks
+
+        try:
+            for rank in range(n_ranks):
+                _, data = reader._load_vt_file(rank)
+                phases = data.get("phases", [])
+                if not phases:
+                    self.__logger.warning(f"No phases found for rank {rank}")
+                    continue
+
+                last_phase = phases[-1]
+
+                if "lb_iterations" in last_phase:
+                    lb_iterations = last_phase["lb_iterations"]
+                    if lb_iterations:
+                        last_lb_iteration = lb_iterations[-1]
+                        iteration_tasks = [task["entity"].get("seq_id", task["entity"].get("id")) for task in last_lb_iteration.get("tasks", [])]
+                        tasks[rank] = iteration_tasks
+                    else:
+                        self.__logger.warning(f"No lb_iterations found in the last phase of rank {rank}")
+                else:
+                    phase_tasks = [task["entity"].get("seq_id", task["entity"].get("id")) for task in last_phase.get("tasks", [])]
+                    tasks[rank] = phase_tasks
+        except (json.JSONDecodeError, KeyError, ValueError, IndexError) as e:
+            self.__logger.error(f"Error processing rank {rank}: {e}")
+            return
+
+        return tasks
+
+    def run(self):
+        parser = argparse.ArgumentParser(description="Extract tasks from JSON files.")
+        parser.add_argument("directory", type=str, help="Directory containing JSON files.")
+        parser.add_argument("--file-stem", type=str, default="data", help="File stem for JSON files (default: 'data').")
+        parser.add_argument("--file-suffix", type=str, default="json", help="File suffix for JSON files (default: 'json').")
+        parser.add_argument("--output", type=str, default="tasks.yml", help="Output YAML file (default: 'tasks.yml').")
+
+        args = parser.parse_args()
+
+        self.__directory = args.directory
```

Comment:
Private member variable would be better defined and initialized (at None) in constructor